app/views/shared/_header-main.html.erb

```diff
@@ -0,0 +1,28 @@
+  
+  <nav class="navbar sticky-top navbar-expand-lg px-3 justify-content-between navbar-dark bg-dark">
+    <div><%= link_to 'le-Club', root_path, class: 'navbar-brand' %><span class='text-muted'>(Nerds Only)</span></div>
+    <div>
+      <button class="navbar-toggler" type="button" data-toggle="collapse" data-target="#navbarNav" aria-controls="navbarNav" aria-expanded="false" aria-label="Toggle navigation">
+        <span class="navbar-toggler-icon"></span>
+      </button>
+      <div class="collapse navbar-collapse navbar-end" id="navbarNav">
+        <ul class="navbar-nav">
+        <% if user_signed_in? %>
+          <li class="nav-item">
+            <%= link_to current_user.name, edit_user_registration_path, class: 'nav-link' %>
+          </li>
+          <li class="nav-item">
+            <%= link_to 'Sign Out', destroy_user_session_path, class: 'nav-link' %>
+          </li>
+        <% else %>
+          <li class="nav-item">
+            <%= link_to 'Sign In', new_user_session_path, class: 'nav-link' %>
+          </li>
+          <li class="nav-item">
+            <%= link_to 'Sign Up', new_user_registration_path, class: 'nav-link' %>
+          </li>
+        <% end %>
```

Comment:
There should be absolutely NO logic in views. View-related logic can be encapsulated in View helpers (app/helpers/*_helper.rb)